server/methods/core/cart.js

```diff
@@ -636,6 +645,7 @@ Meteor.methods({
     // update or insert method
     try {
       Collections.Cart.update(selector, update);
+      Hooks.Events.run("onCartShipmentMethodSet", cart);
```

Comment:
The hook for this should include the shipment method that is being set. This applies to almost all of these cart methods.